cql3/statements/describe_statement.cc

```diff
@@ -0,0 +1,557 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#include "cql3/statements/prepared_statement.hh"
+#include "db/consistency_level_type.hh"
+#include "db/timeout_clock.hh"
+#include "dht/i_partitioner.hh"
+#include "exceptions/exceptions.hh"
+#include "keys.hh"
+#include "partition_slice_builder.hh"
+#include "query-request.hh"
+#include "types.hh"
+#include "cql3/query_processor.hh"
+#include "cql3/cql_statement.hh"
+#include "cql3/statements/raw/describe_statement.hh"
+#include "cql3/statements/describe_statement.hh"
+#include "seastar/core/shared_ptr.hh"
+#include "transport/messages/result_message.hh"
+#include "transport/messages/result_message_base.hh"
+#include "service/query_state.hh"
+#include "data_dictionary/keyspace_metadata.hh"
+#include "data_dictionary/data_dictionary.hh"
+#include <boost/range/adaptor/transformed.hpp>
+#include <boost/range/algorithm/sort.hpp>
+#include <memory>
+#include <optional>
+#include "db/system_keyspace.hh"
+#include "service/storage_proxy.hh"
+#include "cql3/ut_name.hh"
+
+bool is_internal_keyspace(std::string_view name);
+
+namespace cql3 {
+
+namespace statements {
+
+static const sstring KS = "system";
+static const sstring CF = "describe";
+
+/// `describe_statement` provides top level interface for executing describe queries.
+/// All the statements are quering virtual table `cluster_desctiption_tables` or
+/// `elements_description_table` (in db/system_keyspace.cc) which were created
+/// for this purpose and contains descriptions for cluster/elements.
```

Comment:
So if you do a "describe function" on a single function, this also creates the full description of all keyspaces and all tables and everything - potentially thousands of them? Do we find this acceptable always, or just for now (FIXME)?

Reply:
You will only generate description for all of the elements of one keyspace, in other words, if you do `desc function test_ks.func`, you'll generate description for all elements (tables, functions, ...) from `test_ks` keyspace.

I've done some quick local test and created keyspace with around 5k tables (since they have the longest description) and there were no difference in describe execution time